pdf-view-restore.el

```diff
@@ -65,8 +65,15 @@
   "Restore page."
   (when (derived-mode-p 'pdf-view-mode)
     ;; This buffer is in pdf-view-mode
-    (let ((page (pdf-view-restore-get-page)))
-      (when page (pdf-view-goto-page page)))))
+    (let ((page (pdf-view-restore-get-page))
+          (pdf-pages (pdf-info-number-of-pages)))
+      (when page
+        (if (<= page pdf-pages)
+            (pdf-view-goto-page page)
+          (message "Stored page number (%s) in pdf-restore file larger than \
+number of pages in document (%s). Entry has been reset to first page."
+                   page
+                   pdf-pages))))))
```

Comment:
Based on this message, where is the actual logic to reset the entry to the first page?

Reply:
Well, I did not write it down very explicitly, but also in the first commit message I wrote that the "error does not fix itself". With this I mean that, when the error occurs when trying to jump to the non-existent page, then although the document still opens correctly, the package stops updating the pdf-restore entry for the document. Now the when just prevents pdf-restore from trying to jump so that pdf-restore keeps updating the document its entry, and as the docuement opens fine (on page 1), the entry gets updated (also when scrolling through the document).

Also I wrote that it is completely fine if you might want to fix it in a different way, but that otherwise it would be nice if this very simple fix could get merged.